syntax = "proto3";

package storage;

service StorageService {
  rpc Set(SetRequest) returns (SetResponse) {}
  rpc Get(GetRequest) returns (GetResponse) {}
}

message SetRequest {
  string key = 1;
  Value value = 2;
}

message SetResponse {
  oneof result {
    Success success = 1;
    Error error = 2;
  }
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  oneof result {
    Value value = 1;
    Error error = 2;
  }
}

message Value {
  bytes data = 1;
}

message Success {}

message Error {
  string code = 1;
  string message = 2;
}

